keylime/tenant.py

```diff
@@ -642,66 +644,77 @@ def do_status(self, listing=False, returnresponse=False, bulk=False):
                 cert=self.cert,
                 verify=False
             )
+            response = response.json()
         else:
             response = do_cvstatus.get(
                 (f'/v{self.api_version}/agents/{agent_uuid}'),
                 cert=self.cert,
                 verify=False
             )
+            response = response.json()
+            res = response.pop('results')
+            response['results'] = {agent_uuid: res}
 
-        if response.status_code == 503:
```

Comment:
We can't always swap the HTTP status code with the JSON's `code` value because there might not be a JSON payload in certain failures and we still need to handle those.